ops/model.py

```diff
@@ -2508,6 +2872,108 @@ def update_relation_data(self, relation_id: int, _entity: 'UnitOrApplication',
                              key: str, value: str):
         self.relation_set(relation_id, key, value, isinstance(_entity, Application))
 
+    def _run_secret(self, *args: str, return_output: bool = False,
+                    use_json: bool = False) -> Union[str, 'JsonObject', None]:
```

Comment:
Good call -- I've renamed it to `_run_for_secret`. Yeah, it exists solely to map errors to `SecretNotFoundError`, rather than reproducing that try/except in each method.

Regarding logging, apart from `secret-get` (which is it's intended purpose), the hook tools never write potentially sensitive info (secret content) to stdout/stderr. Secret IDs and the like are not sensitive and safe to log.

So I don't think that can happen, but worth a bit of further discussion tomorrow.